diff --git a/dlls/ntdll/unix/unix_private.h b/dlls/ntdll/unix/unix_private.h
index 2f649ca399e..cd490295ee6 100644
--- a/dlls/ntdll/unix/unix_private.h
+++ b/dlls/ntdll/unix/unix_private.h
@@ -724,4 +724,6 @@ static inline int is_gdt_sel( WORD sel )
 
 #endif  /* defined(__i386__) || defined(__x86_64__) */
 
+void DECLSPEC_NOINLINE __wine_dbg_notify_syscall_fault(void);
+
 #endif /* __NTDLL_UNIX_PRIVATE_H */
