diff --git a/dlls/ntdll/ntdll.spec b/dlls/ntdll/ntdll.spec
index 32b1d5e3d6c..0c3bc9a2b46 100644
--- a/dlls/ntdll/ntdll.spec
+++ b/dlls/ntdll/ntdll.spec
@@ -1754,6 +1754,7 @@
 @ extern -private __wine_syscall_dispatcher
 @ extern -private __wine_unix_call_dispatcher
 @ extern -private -arch=arm64ec __wine_unix_call_dispatcher_arm64ec
+@ extern -private -arch=arm64ec __wine_arm64ec_extended_eflags
 @ extern -private __wine_unixlib_handle
 @ stdcall __wine_get_unix_env(ptr ptr long)
 @ stdcall __wine_set_unix_env(ptr ptr)
