diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
index 636b6d7..a3ebe60 100644
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -387,6 +387,10 @@ static UINT64 read_tsc_frequency(void)
     return tsc_frequency;
 }
 
+static void initialize_xstate_features(struct _KUSER_SHARED_DATA *data)
+{
+}
+
 #else
 
 static void initialize_xstate_features(struct _KUSER_SHARED_DATA *data)
